src/cargo/core/resolver/mod.rs

```diff
@@ -256,12 +257,42 @@ impl<'a> Iterator for DepsNotReplaced<'a> {
     }
 }
 
+enum RcList<T> {
+    Data(Rc<(T, RcList<T>)>),
+    Empty,
+}
```

Comment:
Ah not quite, the intention here is that `Clone for RcList<T>` is O(1) wheras `Clone for LinkedList<T>` is O(n)